content/en/apps/guides/updates/preparing-for-4.0.md

```diff
@@ -0,0 +1,214 @@
+---
+title: "Preparing to upgrade to CHT 4.0"
+linkTitle: "Preparing to upgrade to CHT 4.0"
+weight:
+description: >
+  Steps to ensure your CHT App will run smoothly on CHT 4.0 and later
+  
+---
+
+{{% alert title="Note" %}} This guide applies to both self-hosted and Medic hosted deployments. {{% /alert %}}
+
+## Introduction
+
+Medic uses [Semantic Versioning](https://en.wikipedia.org/wiki/Semver#Semantic_versioning) (aka "SemVer") which means that the CHT upgrade from the major 3.x version to the 4.x version denotes there are breaking changes. The key to a successful upgrade will be to understand and plan for these breaking changes. Aside from the Docker hosting infrastructure (out of scope for this prep document), the two breaking changes are around CHT Android and Enketo.
+
+While CHT 4.0 has not been released yet, the effort to be prepared can be quite time consuming, especially for large deployments that may need to do handset upgrades in a worst case.  The sooner deployments start preparing for the upgrade, the easier it will be when it comes to the upgrade itself.  Conveniently, all device and app changes to prepare for 4.x are backwards compatible with 3.x. Prepare now, so you will be ready to upgrade sooner than later!
+
+## CHT Android v1.0.0+ 
+
+This change is straightforward in that CHT 4.x no longer supports versions _before_ `1.0.0`, so deployments need to update their Play Store app. As of this writing, [CHT Android](https://github.com/medic/cht-android/) is at `1.0.4`.  Please see the [Android docs]({{< relref "core/guides/android" >}}) on how to update your app and release it.  Note that Google's Play Store can often have delays which deployments have no control over. Again, the sooner you start, the better.
+
+### Versions in use
+
+After you have published your app, you need to instruct your users to check the Play Store for upgrades. You can then check [CHT Telemetry]({{< relref "apps/guides/performance/telemetry" >}})  to see what CHT Android versions are in use. Assuming you have [couch2pg set up](https://github.com/medic/couch2pg) to pull in your CouchDB data to a PostgreSQL database, this query will list Android versions `count`s for the current year, broken out by `month`, `year` and `version`:
+
+```sql
+select 
```

Comment:
I'm not opposed to this query because it's not wrong.  I think something like this paints a clearer picture:

```
SELECT
	concat(doc#>>'{metadata,year}','-',lpad(doc#>>'{metadata,month}',2,'0')) as telemetry_month, 	
	doc#>>'{device,deviceInfo,app,version}' as cht_android_version,
	count(distinct(doc#>>'{metadata,user}')) AS count_distinct_users,
	count(distinct(doc#>>'{metadata,deviceId}')) AS count_distinct_devices,
	count(*) AS count_telemetry
		
FROM
	couchdb_users_meta 

WHERE
	doc#>>'{type}' = 'telemetry'  
	AND doc#>>'{metadata,year}' = to_char(current_date, 'yyyy')

GROUP BY 
	telemetry_month, cht_android_version	

ORDER BY 
	telemetry_month DESC,
	cht_android_version DESC NULLS LAST
```

- _(Personal preference on the...)_ **Order of columns**, I generally put the least changing and/or the things you are disaggregating by first, and aggregate numbers next
- Includes info where there is **no android version**
- **Highlights number of users** rather than number of telemetry docs (but includes that too)

Reply:
Here's an example of the results of the query above:

<img width="710" alt="Screen Shot 2022-10-05 at 5 37 58 PM" src="https://user-images.githubusercontent.com/10096985/194169474-5a1b4a47-601c-4f46-bca4-19e4a7e1fa2a.png">

Reply:
This is really lovely - thank you @michaelkohn !  Pulling apart the user vs telemetry doc count is especially helpful :heart:

Reply:
omg and distinct devices?!  You really are thoughtful.  Thanks again!